### ISSUE TITLE
[android]: android.os.BadParcelableException

### ISSUE THREAD

**Issue by rtsisyk** [organicmaps/organicmaps#537]
```
java.lang.RuntimeException: 
  at android.app.ActivityThread.callActivityOnStop (ActivityThread.java:4625)
  at android.app.ActivityThread.performStopActivityInner (ActivityThread.java:4595)
  at android.app.ActivityThread.handleStopActivity (ActivityThread.java:4670)
  at android.app.servertransaction.TransactionExecutor.performLifecycleSequence (TransactionExecutor.java:233)
  at android.app.servertransaction.TransactionExecutor.cycleToPath (TransactionExecutor.java:201)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:173)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2017)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:214)
  at android.app.ActivityThread.main (ActivityThread.java:7397)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:492)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:935)
Caused by: android.os.BadParcelableException: 
  at android.os.Parcel.readParcelableCreator (Parcel.java:3042)
  at android.os.Parcel.readParcelable (Parcel.java:2964)
  at android.os.Parcel.readValue (Parcel.java:2866)
  at android.os.Parcel.readArrayMapInternal (Parcel.java:3244)
  at android.os.BaseBundle.initializeFromParcelLocked (BaseBundle.java:292)
  at android.os.BaseBundle.unparcel (BaseBundle.java:236)
  at android.os.BaseBundle.containsKey (BaseBundle.java:509)
  at android.content.Intent.hasExtra (Intent.java:7722)
  at android.app.Activity.onStop (Activity.java:2471)
  at androidx.fragment.app.FragmentActivity.onStop (FragmentActivity.java:520)
  at androidx.appcompat.app.AppCompatActivity.onStop (AppCompatActivity.java:216)
  at com.mapswithme.maps.base.BaseMwmFragmentActivity.onStop (BaseMwmFragmentActivity.java:194)
  at android.app.Instrumentation.callActivityOnStop (Instrumentation.java:1467)
  at android.app.Activity.performStop (Activity.java:8018)
  at android.app.ActivityThread.callActivityOnStop (ActivityThread.java:4617)
  at android.app.ActivityThread.performStopActivityInner (ActivityThread.java:4595)
  at android.app.ActivityThread.handleStopActivity (ActivityThread.java:4670)
  at android.app.servertransaction.TransactionExecutor.performLifecycleSequence (TransactionExecutor.java:233)
  at android.app.servertransaction.TransactionExecutor.cycleToPath (TransactionExecutor.java:201)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:173)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2017)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:214)
  at android.app.ActivityThread.main (ActivityThread.java:7397)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:492)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:935)
```

Nokia 6.1
Android 10 (SDK 29)
21052106.aab

https://play.google.com/console/u/0/developers/8084273541548442467/app/4972992444534608608/vitals/crashes/568790cb/details?days=30

**Comment by rtsisyk** [comment]
Let's wait for more reports. I have no idea how to fix this bug.

**Comment by Jean-BaptisteC** [comment]
Fixed by #5879
